docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+      - [Example - Client drain event](#example---client-drain-event)
+  - [Parameter: `UndiciHeaders`](#parameter-undiciheaders)
+    - [Example 1 - Object](#example-1---object)
+    - [Example 2 - Array](#example-2---array)
+
+## `new Client(url, [options])`
+
+Arguments:
+
+* **url** `URL | string` - It should only include the protocol, hostname, and port.
+* **options** `object` (optional)
+  * **bodyTimeout** `number | null` (optional) - Default: `30e3` -
+  * **headersTimeout** `number | null` (optional) - Default: `30e3` - Node.js v14+ - The amount of time the parser will wait to receive the complete HTTP headers. Defaults to 30 seconds.
+  * **keepAliveMaxTimeout** `number | null` (optional) - Default: `600e3` - The maximum allowed `idleTimeout` when overriden by *keep-alive* hints from the server. Defaults to 10 minutes.
+  * **keepAliveTimeout** `number | null` (optional) - Default: `4e3` - The timeout after which a socket without active requests will time out. Monitors time between activity on a connected socket. This value may be overriden by *keep-alive* hints from the server. Defaults to 4 seconds.
+  * **keepAliveTimeoutThreshold** `number | null` (optional) - Default: `1e3` - A number subtracted from server *keep-alive* hints when overriding `idleTimeout` to account for timing inaccuries caused by e.g. transport latency. Defaults to 1 second.
+  * **maxHeaderSize** `number | null` (optional) - Default: `16384` - The maximum length of request headers in bytes. Defaults to 16KiB.
+  * **pipelining** `number | null` (optional) - Default: `1` - The amount of concurrent requests to be sent over the single TCP/TLS connection according to [RFC7230](https://tools.ietf.org/html/rfc7230#section-6.3.2). Carefully consider your workload and environment before enabling concurrent requests as pipelining may reduce performance if used incorrectly. Pipelining is sensitive to network stack settings as well as head of line blocking caused by e.g. long running requests. Set to `0` to disable keep-alive connections.
+  * **socketPath** `string | null` (optional) - Default: `null` - An IPC endpoint, either Unix domain socket or Windows named pipe.
+  * **tls** `TlsOptions | null` (optional) - Default: `null` - An options object which in the case of `https` will be passed to [`tls.connect`](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback).
+
+Returns: `Client`
+
+### Example - Basic Client instantiation
+
+```js
+'use strict'
+const { Client } = require('undici')
+
+const client = new Client('http://localhost:3000')
+```
+
+## Instance Methods
+
+### `Client.close([ callback ])` 
+
+Closes the client and gracefully waits for enqueued requests to complete before resolving.
+
+Arguments:
+
+* **callback** `(error: Error | null, data: null) => void` (optional)
+
+Returns: `void | Promise<null>` - Only returns a `Promise` if no `callback` argument was passed
+
+#### Example - Request resolves before Client closes
```

Comment:
It's a bit weird that this kind of edge case is the first example.

Reply:
So yes I agree this is not the best first example, but I wanted to keep the API docs alphabetical and then have each method's examples be specific to that method. Thus, if a user lands on this doc page and goes to this doc section for this specific method they will get the exact information they are looking for.

For a more general purpose "how to use" documentation I'd rather have that be its own file or maybe replace our current readme.